prowler/providers/azure/services/entra/entra_service.py

```diff
@@ -19,26 +19,30 @@
 class Entra(AzureService):
     def __init__(self, provider: AzureProvider):
         super().__init__(GraphServiceClient, provider)
-        self.users = asyncio.get_event_loop().run_until_complete(self.__get_users__())
-        self.authorization_policy = asyncio.get_event_loop().run_until_complete(
-            self.__get_authorization_policy__()
-        )
-        self.group_settings = asyncio.get_event_loop().run_until_complete(
-            self.__get_group_settings__()
-        )
-        self.security_default = asyncio.get_event_loop().run_until_complete(
-            self.__get_security_default__()
-        )
-        self.named_locations = asyncio.get_event_loop().run_until_complete(
-            self.__get_named_locations__()
-        )
-        self.directory_roles = asyncio.get_event_loop().run_until_complete(
-            self.__get_directory_roles__()
-        )
-        self.conditional_access_policy = asyncio.get_event_loop().run_until_complete(
-            self.__get_conditional_access_policy__()
+
+        loop = get_event_loop()
+
+        # Get users first alone because it is a dependency for other attributes
+        self.users = loop.run_until_complete(self.__get_users__())
+
+        attributes = loop.run_until_complete(
+            gather(
+                self.__get_authorization_policy__(),
+                self.__get_group_settings__(),
+                self.__get_security_default__(),
+                self.__get_named_locations__(),
+                self.__get_directory_roles__(),
+                self.__get_conditional_access_policy__(),
+            )
         )
 
+        self.authorization_policy = attributes[0]
```

Comment:
Is there any way to get this as a map? Fixed items in list are dangerous sometimes

Reply:
I know that is not the best way to assign, but the only function that is valid in this case is `run_until_complete` because it returns the actual values of attributes and no coroutines like other functions of asyncio. It gives it back in this format, an array ordered with the return values functions. At the moment I can't think of a better way to assign these values.

Reply:
That's fine, if this is know and controlled, thanks @puchy22